python/mxnet/gluon/estimator/estimator.py

```diff
@@ -241,16 +240,35 @@ def fit(self, train_data,
             from a data batch and load into contexts(devices)
         """
 
-
         self.epochs = epochs
-        if not batch_size:
-            batch_size = 32 * len(self.context)
+        if isinstance(train_data, gluon.data.DataLoader):
+            num_batches = len(train_data)
+            total_samples = len(train_data._dataset)
+            if total_samples == 0:
+                raise ValueError("DataLoader is Empty. Please refer to gluon.data.DataLoader "
+                                 "for more detail")
+            for dt, _ in train_data:
```

Comment:
this will error out if  train data is a tuple of 3 or more, get it from batch[0].shape[0], it's safer to assume first of batch is the data, batch[1], batch[2] can be label1 label2 for multi label classifications.

Reply:
True. modified the code. [5f94478](https://github.com/apache/incubator-mxnet/pull/14587/commits/5f9447892bfc453963fe6765f171efb556a12649)